pkg/codegen/docs/templates/header.tmpl

```diff
@@ -1,7 +1,8 @@
 {{ define "header" }}
 ---
 title: "{{ .Title }}"
-block_external_search_index: true
+title_tag: "{{ .TitleTag }}"
```

Comment:
@cnunciato are we ready to remove the `block_external_search_index`?

Reply:
With this PR, we should be, yes. I'll take a look at this this morning.

Reply:
Ok in that case, @sean1588 we'll need to remove it from the `index.tmpl` template as well. 

Speaking of `index.tmpl`, I am also wondering if we should do a similar improvement to the meta description there as well? If it is ok to punt on that for now, I am ok with that. Just wondering...

Reply:
I was assuming we would do that, so would prefer it if it didn't require too much more effort.

Reply:
At the very least, we do need to remove the `block` properties from the frontmatter in the index files, though, yeah.

Reply:
OK, @cnunciato and @praneetloke , added the changes for the title and description for the index pages. The meta_description on the modules index page is not the greatest, right now it is just essentially a list of the resources in that module, not sure what else we can give it as there is no human readable description at the module level.... The meta_description on the package index page looks fine though.

Reply:
When I run the generator, I see this fairly often, which doesn't seem right:

![image](https://user-images.githubusercontent.com/274700/79621944-2a523080-80ca-11ea-80e6-523c6ae74e38.png)

Looking at the `meta_desc`s that do render, though, I do think we could probably still be a bit more descriptive. In the issue, I mention referring to the API docs as a guide. They currently render:

```
<meta name="description" content="Explore members of the s3 module in the @pulumi/aws package.">
```

So I think beyond just "Provides a S3 bucket resource", we could do something similar:

```
<meta name="description" content="Explore the resources and functions of the s3 module in the aws package.">
```

And at the resource level, maybe something like:

```
<meta name="description" content="Explore the Bucket resource of the aws/s3 module, including examples, input properties, output properties, lookup functions, and supporting types.">
```

It's just a bit of extra text, but this is what gets used in previews and possibly search results, so it's good to try to put something informative in that space. For example:

![image](https://user-images.githubusercontent.com/274700/79622634-e57bc900-80cc-11ea-8a79-a3c9a8bb8dec.png)